### PR TITLE
Use createMock() instead of a getter

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

Before I go on, are you interested in such change @nicolas-grekas ?

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Sure, please merge in 4.4 and up to 5.x also, and fix other branches while doing so.

File: src/Symfony/Bundle/WebProfilerBundle/Tests/Profiler/TemplateManagerTest.php [hunk 1]
```diff
@@ -121,13 +114,6 @@ protected function mockTwigEnvironment()
 
         return $this->twigEnvironment;
     }
-
-    protected function mockProfiler()
-    {
-        $this->profiler = $this->createMock(Profiler::class);
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/Profiler/TemplateManagerTest.php:127]: we should remove the property then (and not use it anymore)

Comment (OskarStark) [reply src/Symfony/Bundle/WebProfilerBundle/Tests/Profiler/TemplateManagerTest.php:127]: Ah it is used by refernce in a test:
```php
    /**
     * if template exists in both profile and profiler then its name should be returned.
     */
    public function testGetNameValidTemplate()
    {
        $this->profiler->expects($this->any())
            ->method('has')
            ->withAnyParameters()
            ->willReturnCallback([$this, 'profilerHasCallback']);

        $this->assertEquals('FooBundle:Collector:foo.html.twig', $this->templateManager->getName(new ProfileDummy(), 'foo'));
    }
```

**Review by jderusse — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Comment by chalasr** [comment]
Thanks @OskarStark.

**Pull request merged**